### PR TITLE
PROD-913 Post Data getting lost if user edit any other post while creating new post. #1494

### LINKED ISSUE #1494: Post Data getting lost if user edit any other post while creating new post.
**Describe the bug**

UX Bug - Post Data getting lost if user edit any other post while creating new post.
UX- In this scenario any author will loss typed content in text editor if they edited any post at the same time.
Refer video for details.

**To Reproduce**

https://www.loom.com/share/f415bee7f1654c8c850c900ef7528259

**Expected behavior**

This should not remove data if user edit old activity at the time of writing new post.

**Screenshots**

https://www.loom.com/share/f415bee7f1654c8c850c900ef7528259

**Support ticket links**

N/A- Internal from QA.

**Jira issue** : [PROD-913]

**Theme Issue:**
https://github.com/buddyboss/buddyboss-theme/issues/979

[PROD-913]: https://buddyboss.atlassian.net/browse/PROD-913?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by shohelbuddyboss**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1494 .

### Changelog entry

> If there any draft post, edit activity post will not working until the main post content delete or publish.

### CODE REVIEW

File: src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js [hunk 1]
```diff
@@ -42,6 +42,8 @@ window.bp = window.bp || {};
 			this.postFormView();
 
 			this.postFormPlaceholderView();
+
+			this.checkDraftContentInPostForm();
```
Comment (mehulkaklotar) [inline src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js:46]: @shohelbuddyboss We should make use of Backbone events here like this https://github.com/buddyboss/buddyboss-platform/blob/dev/src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js#L2505

https://github.com/buddyboss/buddyboss-platform/blob/dev/src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js#L2532

File: src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js [hunk 2]
```diff
@@ -76,6 +78,92 @@ window.bp = window.bp || {};
...
+		checkDraftAttachments: function(){
+
+			if ( this.postForm.$el.hasClass( 'bp-activity-edit' ) ){
+				return;
+			}
+
+			var hasDraft;
+
+			var $whatsNewAttachments = $( '#whats-new-attachments' );
+			var countAttachmets = $whatsNewAttachments.find( '.dz-preview' ).length;
```
Comment (mehulkaklotar) [inline src/bp-templates/bp-nouveau/js/buddypress-activity-post-form.js:121]: @shohelbuddyboss can you check this without `dz-preview` class elements length and make use of dropzone objects to find if any media and document is added or not in the object?

**Review by mehulkaklotar — COMMENTED** [review]
@shohelbuddyboss you will need different approach here to fix this. There should be one global variable attached to PostForm object and then there should be one wrapper function to check all details if post form has draft or not. This implementation will not work. 

Let's have discussion about this change before you proceed with it.

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@shohel Need to make the changes as mehul suggested.

**Comment by jitendrabanjara1991** [comment]
@chetansatasiya 
As per request changes, I have changed code and some issue with draft content so fixed it.
Here is the video for the proof - https://screencast-o-matic.com/watch/crQnl1VQOIL
@sourabhmatolia can you send check alert text which displaying as popup? Text something like this - **You have a draft post, publish the draft post first in order to edit any activity.**
Please let me know need to change anything else
Thanks.

**Pull request closed without merging**